drivers/spi/spi_gecko.c

```diff
@@ -0,0 +1,329 @@
+
+/* Structure Declarations */
+
+struct spi_gecko_data {
+	struct spi_context ctx;
+};
+
+struct spi_gecko_config {
+	USART_TypeDef *base;
+	CMU_Clock_TypeDef clock;
+	struct soc_gpio_pin pin_rx;
+	struct soc_gpio_pin pin_tx;
+	struct soc_gpio_pin pin_clk;
+	u8_t loc_rx;
+	u8_t loc_tx;
+	u8_t loc_clk;
+};
+
+
+/* Helper Functions */
+static int spi_config(struct device *dev, const struct spi_config *config,
+		      u16_t *control)
+{
+	const struct spi_gecko_config *gecko_config = dev->config->config_info;
+	struct spi_gecko_data *data = DEV_DATA(dev);
+
+	if (SPI_WORD_SIZE_GET(config->operation) != SPI_WORD_SIZE) {
+		LOG_ERR("Word size must be %d", SPI_WORD_SIZE);
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_CS_ACTIVE_HIGH) {
+		LOG_ERR("CS active high not supported");
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_LOCK_ON) {
+		LOG_ERR("Lock On not supported");
+		return -ENOTSUP;
+	}
+
+	if ((config->operation & SPI_LINES_MASK) != SPI_LINES_SINGLE) {
+		LOG_ERR("Only supports single mode");
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_TRANSFER_LSB) {
+		LOG_ERR("LSB first not supported");
+		return -ENOTSUP;
+	}
+
+	if (config->operation & (SPI_MODE_CPOL | SPI_MODE_CPHA)) {
+		LOG_ERR("Only supports CPOL=CPHA=0");
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_OP_MODE_SLAVE) {
+		LOG_ERR("Slave mode not supported");
+		return -ENOTSUP;
+	}
+
+	/* Set Loopback */
+	if (config->operation & SPI_MODE_LOOP) {
+		gecko_config->base->CTRL |= USART_CTRL_LOOPBK;
+	} else {
+		gecko_config->base->CTRL &= ~USART_CTRL_LOOPBK;
+	}
+
+	/* Set word size */
+	gecko_config->base->FRAME = usartDatabits8
+	    | USART_FRAME_STOPBITS_DEFAULT
+	    | USART_FRAME_PARITY_DEFAULT;
+
+	/* At this point, it's mandatory to set this on the context! */
+	data->ctx.config = config;
+
+	spi_context_cs_configure(&data->ctx);
+
+	return 0;
+}
+
+static void spi_gecko_send(USART_TypeDef *usart, u8_t frame)
+{
+	/* Write frame to register */
+	USART_Tx(usart, frame);
+
+	/* Wait until the transfer ends */
+	while (!(usart->STATUS & USART_STATUS_TXC))
+		;
+}
+
+static u8_t spi_gecko_recv(USART_TypeDef *usart)
+{
+	/* Return data inside rx register */
+	return (u8_t)usart->RXDATA;
+}
+
+static bool spi_gecko_transfer_ongoing(struct spi_gecko_data *data)
+{
+	return spi_context_tx_on(&data->ctx) || spi_context_rx_on(&data->ctx);
+}
+
+static inline u8_t spi_gecko_next_tx(struct spi_gecko_data *data)
+{
+	u8_t tx_frame = 0;
+
+	if (spi_context_tx_buf_on(&data->ctx)) {
+		tx_frame = UNALIGNED_GET((u8_t *)(data->ctx.tx_buf));
+	}
+
+	return tx_frame;
+}
+
+static int spi_gecko_shift_frames(USART_TypeDef *usart,
+				  struct spi_gecko_data *data)
+{
+	u8_t tx_frame;
+	u8_t rx_frame;
+
+	tx_frame = spi_gecko_next_tx(data);
+	spi_gecko_send(usart, tx_frame);
+	spi_context_update_tx(&data->ctx, 1, 1);
+
+	rx_frame = spi_gecko_recv(usart);
+	if (spi_context_rx_buf_on(&data->ctx)) {
```

Comment:
tiny style issue, get an empty line before the if (it's valid if only the previously assigned variable is tested in the if)

Reply:
done